pkg/operator/helpers/helpers.go

```diff
@@ -724,6 +725,39 @@ func GetHubKubeconfig(ctx context.Context,
 	}
 }
 
+// RemoveWorkConfigSecret is used to remove the secret of work config from the cluster manager namespace.
+func RemoveWorkConfigSecret(ctx context.Context,
+	operatorClient kubernetes.Interface,
+	clusterManagerNamespace string,
+) error {
+	if _, err := operatorClient.CoreV1().Namespaces().Get(ctx, clusterManagerNamespace, metav1.GetOptions{}); err != nil {
+		if errors.IsNotFound(err) {
+			return nil
+		}
+		return err
+	}
+	err := operatorClient.CoreV1().Secrets(clusterManagerNamespace).Delete(ctx, WorkDriverConfig, metav1.DeleteOptions{})
+	if err != nil && errors.IsNotFound(err) {
+		return nil
+	}
+
+	return err
+}
+
+// GetOperatorNamespace is used to get the namespace where the operator is running.
+func GetOperatorNamespace() (string, error) {
```

Comment:
I thought we already get the namespace from the controllerContext

Reply:
the work-driver-config secret is supposed to be created in the namespace (`open-cluster-management` by default) as the operator, not in the operand namespace (`open-cluster-management-hub` by default), because the operand namespace may not exist before clustermanager is created.

Reply:
yes, but you can get the namespace from controllerContext which get the namespace the same as what this func does.

Reply:
oh... I see, it's in command ControllerContext: https://github.com/openshift/library-go/blob/0f907b20fed14da2731dbb6a537e97c691579dc7/pkg/controller/controllercmd/builder.go#L41-L60 
updated this.